homeassistant/components/device_tracker/owntracks.py

```diff
@@ -196,70 +201,70 @@ def owntracks_event_update(topic, payload, qos):
+                _set_gps_from_zone(kwargs, location, zone)
+
+            hass.async_add_job(async_see(**kwargs))
+            async_see_beacons(dev_id, kwargs)
 
+        @callback
         def leave_event():
             """Execute leave event."""
-            with LOCK:
-                regions = REGIONS_ENTERED[dev_id]
-                if location in regions:
-                    regions.remove(location)
-                new_region = regions[-1] if regions else None
-
-                if new_region:
-                    # Exit to previous region
-                    zone = hass.states.get(
-                        "zone.{}".format(slugify(new_region)))
-                    _set_gps_from_zone(kwargs, new_region, zone)
-                    _LOGGER.info("Exit to %s", new_region)
-                    see(**kwargs)
-                    see_beacons(dev_id, kwargs)
-
-                else:
-                    _LOGGER.info("Exit to GPS")
-                    # Check for GPS accuracy
-                    valid_gps = True
-                    if 'acc' in data:
-                        if data['acc'] == 0.0:
-                            valid_gps = False
-                            _LOGGER.warning(
-                                'Ignoring GPS in region exit because accuracy'
-                                'is zero: %s',
-                                payload)
-                        if (max_gps_accuracy is not None and
-                                data['acc'] > max_gps_accuracy):
-                            valid_gps = False
-                            _LOGGER.info(
-                                'Ignoring GPS in region exit because expected '
-                                'GPS accuracy %s is not met: %s',
-                                max_gps_accuracy, payload)
-                    if valid_gps:
-                        see(**kwargs)
-                        see_beacons(dev_id, kwargs)
-
-                beacons = MOBILE_BEACONS_ACTIVE[dev_id]
-                if location in beacons:
-                    beacons.remove(location)
-                    _LOGGER.info("Remove beacon %s", location)
+            regions = regions_entered[dev_id]
+            if location in regions:
+                regions.remove(location)
+            new_region = regions[-1] if regions else None
+
+            if new_region:
+                # Exit to previous region
+                zone = hass.states.get(
+                    "zone.{}".format(slugify(new_region)))
+                _set_gps_from_zone(kwargs, new_region, zone)
+                _LOGGER.info("Exit to %s", new_region)
+                hass.async_add_job(async_see(**kwargs))
+                async_see_beacons(dev_id, kwargs)
+
+            else:
+                _LOGGER.info("Exit to GPS")
+                # Check for GPS accuracy
+                valid_gps = True
+                if 'acc' in data:
+                    if data['acc'] == 0.0:
```

Comment:
Totally realize you didn't write this in this PR, but seems like it would be [better in a constant](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants) named `NO_GPS_ACCURACY` or something